beesdoo_product/models/beesdoo_product.py

```diff
@@ -247,16 +265,45 @@ def _unit_same_category(self):
                 )
 
     @api.multi
-    @api.depends("seller_ids")
+    @api.depends("seller_ids", "supplier_taxes_id", "taxes_id")
     def _compute_cost(self):
         for product in self:
             suppliers = product._get_main_supplier_info()
             if len(suppliers) > 0:
+                price = suppliers[0].price
+                supplier_taxes = product.supplier_taxes_id.filtered(
+                    lambda t: t.amount_type == "percent" and t.price_include
+                )
+                supplier_taxes_factor = 1 / (
+                    1 + sum(supplier_taxes.mapped("amount")) / 100
+                )
+                sale_taxes = product.taxes_id.filtered(
+                    lambda t: t.amount_type == "percent" and t.price_include
+                )
+                sale_taxes_factor = 1 + sum(sale_taxes.mapped("amount")) / 100
+                profit_margin_supplier = suppliers[0].name.profit_margin
+                profit_margin_product_category = suppliers[
+                    0
+                ].product_tmpl_id.categ_id.profit_margin
+                profit_margin = (
+                    profit_margin_supplier or profit_margin_product_category
+                )
```

Comment:
Half of this code, including database call will never apply on a database considering suggested_price_reference. But we perform both options on each loop.

The code should go inside the if/else statement...

Reply:
I think you're reasoning here is "depending on the value of `suggested_price_reference` either `supplier_taxes_factor` or `sale_taxes_factor` is computed and used."
It's actually: "depending on the value of `suggested_price_reference` the profit_margin_factor is computed differently from the profit_margin, and independent of that both `supplier_taxes_factor` or `sale_taxes_factor` are always computed and used."
So I don't think a change is required here.

Reply:
Ok I got it ! Could hence add some comment to make it easier to dig in ? thanks !